### PR TITLE
Config Documentation for Pebblo

### PR DESCRIPTION
**Pull request by KunalJadhav5**
## Added Markdown Document for Configuration

-  Created config.md

image

-  Updated Installation.md

image

-  Added Configuration in Introduction 

image

### CODE REVIEW

File: docs/gh_pages/docs/config.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+### Introduction
+
+This configuration file specifies settings for various components of the Pebblo.
+
+### Configuration Details
+
+#### Daemon
+
+- `port`: Specifies the port number on which the Pebblo daemon listens for incoming connections.
+- `host`: Specifies the host address on which the Pebblo daemon is running.
```
Comment (shreyas-damle) [inline docs/gh_pages/docs/config.md:12]: We should add note that if we change these values then we need to set `PEBBLO_CLASSIFIER_URL ` for PebbloSafeDataLoader to send requests to the changes host:port.

File: docs/gh_pages/docs/config.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+### Logging
+
+- `level`: Sets the logging level for the Pebblo application. Possible values are 'info', 'debug', 'error', 'warning', and 'critical'.
+
+### Reports
+
+- `format`: Specifies the format of generated reports. Available options include 'pdf', 'html', etc.
```
Comment (shreyas-damle) [inline docs/gh_pages/docs/config.md:20]: We are only supporting `pdf` right now.

File: docs/gh_pages/docs/config.md (hunk 1 above)
Comment (shreyas-damle) [inline docs/gh_pages/docs/config.md:12]: `on which the Pebblo daemon is running.` => `on which the Pebblo daemon to run.`

File: docs/gh_pages/docs/config.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  format: pdf
+  renderer: weasyprint
+  outputDir: ~/.pebblo
+
+```
+
+### Validation
+
+- The Pebblo application validates the configuration file upon startup.
+- If the configuration file contains errors or invalid values, Pebblo will log appropriate error messages and may use default values for parameters.
```
Comment (shreyas-damle) [inline docs/gh_pages/docs/config.md:42]: `may` => `will`. Please confirm the behaviour by running with wrong values.

File: docs/gh_pages/docs/config.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+```commandline
+daemon:
+  port: 8000
+  host: localhost
+logging:
+  level: info
+reports:
+  format: pdf
+  renderer: weasyprint
+  outputDir: ~/.pebblo
```
Comment (shreyas-damle) [inline docs/gh_pages/docs/config.md:35]: We can mention that these all are default values.

File: docs/gh_pages/docs/introduction.md [hunk 5]
```diff
@@ -31,5 +31,6 @@ Pebblo has two components.
 - [Development Environment](development.md)
 - [Pebblo Daemon](daemon.md)
 - [Pebblo Safe DataLoader for Langchain](rag.md)
+- [Pebblo Configuration File ](config.md)
```
Comment (shreyas-damle) [inline docs/gh_pages/docs/introduction.md:34]: @srics Is this fine if we put this as new section?

**Review by shreyas-damle — CHANGES_REQUESTED** [review]

**Review by Raj725 — DISMISSED** [review]
LGTM

File: docs/gh_pages/docs/config.md (hunk 2 above)
Comment (KunalJadhav5) [reply docs/gh_pages/docs/config.md]: Fixed.

@shreyas-damle

File: docs/gh_pages/docs/config.md (hunk 1 above)
Comment (KunalJadhav5) [reply docs/gh_pages/docs/config.md]: Fixed

@shreyas-damle

File: docs/gh_pages/docs/config.md (hunk 3 above)
Comment (KunalJadhav5) [reply docs/gh_pages/docs/config.md]: Fixed.

File: docs/gh_pages/docs/config.md (hunk 4 above)
Comment (KunalJadhav5) [reply docs/gh_pages/docs/config.md:35]: Fixed.

**Review by shreyas-damle — APPROVED** [review]

**Pull request merged**